frontend/templates/product-list/MetaTags.tsx

```diff
@@ -104,9 +121,12 @@ function useCanonicalUrl(
    const itemTypeHandle = itemType
       ? `/${encodeURIComponent(stylizeDeviceItemType(itemType))}`
       : '';
+   const variant = useVariant();
 
    return `${appContext.ifixitOrigin}${productListPath(
-      productList
+      productList,
+      undefined,
+      productList.indexVariantsInsteadOfDevice ? variant : undefined
    )}${itemTypeHandle}${page > 1 ? `?${PRODUCT_LIST_PAGE_PARAM}=${page}` : ''}`;
```

Comment:
I don't see a reason that we don't pass the item type to `productListPath`, and instead compute it on our own. I'm just observing, not saying you need to change it in this pull.